src/coreclr/src/pal/src/arch/arm64/dispatchexceptionwrapper.S

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// ==++==
+//
+
+// ==--==
+//
+// Implementation of the PAL_DispatchExceptionWrapper that is
+// interposed between a function that caused a hardware fault
+// and PAL_DispatchException that throws an SEH exception for
+// the fault, to make the stack unwindable.
+//
+
+#include "unixasmmacros.inc"
+
+// Offset of the return address from the PAL_DispatchException in the PAL_DispatchExceptionWrapper
+.globl C_FUNC(PAL_DispatchExceptionReturnOffset)
+C_FUNC(PAL_DispatchExceptionReturnOffset):
+    .int LOCAL_LABEL(PAL_DispatchExceptionReturn) - C_FUNC(PAL_DispatchExceptionWrapper)
+
+//
+// PAL_DispatchExceptionWrapper will never be called; it only serves
+// to be referenced from a stack frame on the faulting thread.  Its
+// unwinding behavior is equivalent to any standard function.
+// It is analogous to the following source file.
+//
+// extern "C" void PAL_DispatchException(CONTEXT *pContext, EXCEPTION_RECORD *pExceptionRecord, MachExceptionInfo *pMachExceptionInfo);
+//
+// extern "C" void PAL_DispatchExceptionWrapper()
```

Comment:
We will need to verify that stack unwinding in LLDB is correct when the PAL_DispatchExceptionWrapper is on the stack once we can execute managed code.

Reply:
LGTM
```
  * frame #0: 0x0000000101308fc4 libcoreclr.dylib`VirtualCallStubManager::predictStubKind(this=0x000000010281d810, stubStartAddress=15942320479165980345) at virtualcallstubcpu.hpp:600:28
    frame #1: 0x0000000101246bf8 libcoreclr.dylib`VirtualCallStubManager::getStubKind(this=0x000000010281d810, stubStartAddress=15942320479165980345, usePredictStubKind=YES) at virtualcallstub.h:343:57
    frame #2: 0x000000010130daac libcoreclr.dylib`VirtualCallStubManager::FindStubManager(stubAddress=15942320479165980345, wbStubKind=0x00000001006969b4, usePredictStubKind=YES) at virtualcallstub.cpp:987:18
    frame #3: 0x000000010152e5bc libcoreclr.dylib`AdjustContextForVirtualStub(pExceptionRecord=0x0000000000000000, pContext=0x0000000103017a00) at stubs.cpp:1155:5
    frame #4: 0x00000001012bc08c libcoreclr.dylib`Thread::VirtualUnwindToFirstManagedCallFrame(pContext=0x0000000103017a00) at stackwalk.cpp:729:13
    frame #5: 0x0000000101533b78 libcoreclr.dylib`HandleHardwareException(ex=0x0000000100697608) at exceptionhandling.cpp:5114:9
    frame #6: 0x000000010101b7e0 libcoreclr.dylib`SEHProcessException(exception=0x0000000100697608) at seh.cpp:267:21
    frame #7: 0x00000001010b41d8 libcoreclr.dylib`::PAL_DispatchException(pContext=0x0000000100697c60, pExRecord=0x0000000100697bc8, pMachExceptionInfo=0x0000000100697688) at machexception.cpp:388:29
    frame #8: 0x00000001010b36ec libcoreclr.dylib`PAL_DispatchExceptionWrapper at dispatchexceptionwrapper.S:43
    frame #9: 0x00000002801f9eb8
    frame #10: 0x0000000101343098 libcoreclr.dylib`CallDescrWorkerWithHandler(pCallDescrData=0x000000016fdfef90, fCriticalCall=NO) at callhelpers.cpp:68:5
    frame #11: 0x0000000101343bd8 libcoreclr.dylib`MethodDescCallSite::CallTargetWorker(this=0x000000016fdff210, pArguments=0x000000016fdff3d0, pReturnValue=0x0000000000000000, cbReturnValue=0) at callhelpers.cpp:544:9
    frame #12: 0x00000001010d186c libcoreclr.dylib`MethodDescCallSite::Call(this=0x000000016fdff210, pArguments=0x000000016fdff3d0) at callhelpers.h:458:9
    frame #13: 0x000000010118afa8 libcoreclr.dylib`CorHost2::CreateAppDomainWithManager(this=0x000000010030b270, wszFriendlyName=u"unixcorerun", dwFlags=336, wszAppDomainManagerAssemblyName=0x0000000000000000, wszAppDomainManagerTypeName=0x0000000000000000, nProperties=6, pPropertyNames=0x0000000100308250, pPropertyValues=0x0000000100308290, pAppDomainID=0x000000016fdff67c) at corhost.cpp:627:15
    frame #14: 0x00000001010be33c libcoreclr.dylib`::coreclr_initialize(exePath="/Users/steve/git/runtime/artifacts/tests/coreclr/OSX.arm64.Debug/Tests/Core_Root/corerun", appDomainFriendlyName="unixcorerun", propertyCount=6, propertyKeys=0x000000016fdff7a8, propertyValues=0x000000016fdff778, hostHandle=0x000000016fdff680, domainId=0x000000016fdff67c) at unixinterface.cpp:239:16
    frame #15: 0x000000010000426c corerun`ExecuteManagedAssembly(currentExeAbsolutePath="/Users/steve/git/runtime/artifacts/tests/coreclr/OSX.arm64.Debug/Tests/Core_Root/corerun", clrFilesAbsolutePath="/Users/steve/git/runtime/artifacts/tests/coreclr/OSX.arm64.Debug/Tests/Core_Root", managedAssemblyAbsolutePath="/Users/steve/git/x64.console/obj/Debug/net5.0/x64.console.dll", managedAssemblyArgc=0, managedAssemblyArgv=0x0000000000000000) at coreruncommon.cpp:490:22
    frame #16: 0x0000000100002da4 corerun`main(argc=2, argv=0x000000016fdff9d8) at corerun.cpp:148:20
    frame #17: 0x00000001d9af6844 libdyld.dylib`start + 4
(lldb) frame
```